### ISSUE TITLE
Combining VarStores

### ISSUE THREAD

**Issue by jafioti** [LaurentMazare/tch-rs#389]
I have a language model I pretrained on some data. I want to then use part of this model (the transformer) as part of another model. I'm having trouble figuring out the proper way to do this, as I end up having two varstores, one for loading the transformer model's weights, and one for the rest of the final model.

The problem with this is that now this can't be saved as one model, nor can it be optimized with a single optimizer. Is there a way to combine varstores together?

**Comment by LaurentMazare** [comment]
Maybe it could be possible to use a single varstore for your whole model? This would make it easier to save and run an optimizer on the whole thing. One tricky bit would be to load the pre-trained weights for the transformer part but for this you can probably use [load_partial](https://docs.rs/tch/0.5.0/tch/nn/struct.VarStore.html#method.load_partial), this function does not fail if some variables in the varstore are not available in the weights file.

**Comment by jafioti** [comment]
Yes I tried to use load_partial, but the issue is that the paths don't match up. Is it possible to get a varstore of only certian paths in the varstore? Because if I add a path to a varstore, I get an nn::Path, which does not have a copy function. 

Is it possible to store the variables in each module, rather than have a single large varstore for the entire model? This seems to be how the pytorth python API does it, though I don't know how they do it in libtorch.

**Comment by jafioti** [comment]
I am kind of getting around the solution by loading my pretrained model seperately and copying the appropriate tensors over to the main model by doing:
```
dest = source.detach().copy();
```
Does this mean that the dest will continue to be tracked? Is there a better way to copy a tensor over to a tensor being tracked?

**Comment by LaurentMazare** [comment]
I'm not sure to understand what you mean by "tracked" here. You can still backpropagate on `dest` after this block if that's what you're asking, the `no_grad` block only prevents gradients from being retained for the closure's duration.

**Comment by jafioti** [comment]
Ok that's what I'm asking. It doesn't seem like the most optimal way, but currently just loading each model separately and copying the variables from the pretrained to the new one seems to work for me. It also seems the most intuitively straightforward.

**Comment by dbsxdbsx** [comment]
@LaurentMazare , one side question. Does the below 2 equivalent:
```rust
let train_images = tch::no_grad(|| dataset.train_images.apply_t(&net, false));
let train_images = tch::no_grad(|| dataset.train_images.apply(&net));
```
